candle-examples/examples/qwen/main.rs

```diff
@@ -85,6 +88,10 @@ impl TextGeneration {
             Some(token) => token,
             None => anyhow::bail!("cannot find the <|endoftext|> token"),
         };
+        let eos_token2 = match self.tokenizer.get_token("<|im_end|>") {
```

Comment:
If this applies to the old qwen models too, did you check that all of them have this token in their tokenizers? If not we should probably keep an `Option<u32>` instead.

Reply:
Great thanks!